### PR TITLE
feat: Change OPCM salt to include user supplied arg

### PR DESCRIPTION
**Pull request by blmalone**

### CODE REVIEW

**Comment by blmalone** [comment]
* **#12161** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/12161?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12156** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/12156?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12153** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/12153?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12152** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/12152?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @blmalone and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/contracts-bedrock/scripts/DeployOPChain.s.sol [hunk 1]
```diff
@@ -145,6 +152,12 @@ contract DeployOPChainInput is BaseDeployIO {
         DeployUtils.assertImplementationSet(address(_opcmProxy));
         return _opcmProxy;
     }
+
+    function saltMixer() public view returns (string memory) {
+        // TODO: Temporarily allowing empty 'saltMixer' to be provided.
+        // require(bytes(_saltMixer).length != 0, "DeployOPChainInput: not set");
```
Comment (mds1) [inline packages/contracts-bedrock/scripts/DeployOPChain.s.sol]: Any reason not to permanently allow this? It seems ok to me if saltMixer is empty

File: packages/contracts-bedrock/src/L1/OPContractsManager.sol [hunk 2]
```diff
@@ -359,6 +372,10 @@ contract OPContractsManager is ISemver, Initializable {
         if (_input.roles.unsafeBlockSigner == address(0)) revert InvalidRoleAddress("unsafeBlockSigner");
         if (_input.roles.proposer == address(0)) revert InvalidRoleAddress("proposer");
         if (_input.roles.challenger == address(0)) revert InvalidRoleAddress("challenger");
+
+        if (_input.startingAnchorRoots.length == 0) revert InvalidStartingAnchorRoots();
+        // TODO: Temporarily allowing empty 'saltMixer' to be provided.
+        // if (bytes(_saltMixer).length == 0) revert InvalidSaltMixer();
```
Comment (mds1) [inline packages/contracts-bedrock/src/L1/OPContractsManager.sol]: similar comment
```suggestion
        if (_input.startingAnchorRoots.length == 0) revert InvalidStartingAnchorRoots();
```

File: packages/contracts-bedrock/src/L1/OPContractsManager.sol [hunk 3]
```diff
@@ -187,6 +189,12 @@ contract OPContractsManager is ISemver, Initializable {
     /// @notice Thrown when the latest release is not set upon initialization.
     error LatestReleaseNotSet();
 
+    /// @notice Thrown when the starting anchor roots are not provided.
+    error InvalidStartingAnchorRoots();
+
+    /// @notice Thrown when the salt mixer is not provided.
+    error InvalidSaltMixer();
```
Comment (mds1) [inline packages/contracts-bedrock/src/L1/OPContractsManager.sol]: and if above suggestion is applied we can also delete this
```suggestion
```

**Review by mds1 — APPROVED** [review]
LGTM!

File: packages/contracts-bedrock/scripts/DeployOPChain.s.sol (hunk 1 above)
Comment (blmalone) [reply packages/contracts-bedrock/scripts/DeployOPChain.s.sol]: https://github.com/ethereum-optimism/optimism/pull/12152/commits/012cdf54a9504bd6b62f9e6eab4acce452642193

File: packages/contracts-bedrock/src/L1/OPContractsManager.sol (hunk 2 above)
Comment (blmalone) [reply packages/contracts-bedrock/src/L1/OPContractsManager.sol]: https://github.com/ethereum-optimism/optimism/pull/12152/commits/012cdf54a9504bd6b62f9e6eab4acce452642193

File: packages/contracts-bedrock/src/L1/OPContractsManager.sol (hunk 3 above)
Comment (blmalone) [reply packages/contracts-bedrock/src/L1/OPContractsManager.sol]: https://github.com/ethereum-optimism/optimism/pull/12152/commits/012cdf54a9504bd6b62f9e6eab4acce452642193

**Comment by blmalone** [comment]
### Merge activity

* **Sep 26, 3:10 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/12152) disabled "merge when ready" on this PR due to: a merge conflict with the target branch; resolve the conflict and try again..

File: op-chain-ops/deployer/opcm/opchain.go [hunk 4]
```diff
@@ -34,6 +34,7 @@ type DeployOPChainInput struct {
 	BlobBaseFeeScalar uint32
 	L2ChainId         *big.Int
 	OpcmProxy         common.Address
+	SaltMixer         string
```
Comment (mslipper) [inline op-chain-ops/deployer/opcm/opchain.go:37]: op-deployer currently generates a `create2` salt on startup - could you wire this up with that?

**Review by mslipper — COMMENTED** [review]
lgtm pending comment

File: op-chain-ops/deployer/opcm/opchain.go (hunk 4 above)
Comment (blmalone) [reply op-chain-ops/deployer/opcm/opchain.go:37]: Using `st.Create2Salt` as the initial value of `SaltMixer`: https://github.com/ethereum-optimism/optimism/pull/12152/commits/093331de1baf4f610e8169e73950c3490e557fc3

**Review by mslipper — APPROVED** [review]
lgtm

**Pull request merged**